### PR TITLE
Feature/calendar view 

### PR DESCRIPTION
**Pull request by Yuecheng-Rong**
## Description

Commit the spike prototype for full-calendar view. 
No manual testing document uploaded 

## Related Issue or Trello Card

[Trello card](https://trello.com/c/KGnkdIvE/24-spike-prototype-for-calendar-view)

## Motivation and Context

Generate events data (including both title and date) from slide store and publish in the calendar view. Please notice that only slides with correct format could be generated.
Prototype:
On laptop:
screen shot 2018-03-01 at 9 44 43 pm
On Phone (emulator):
screen shot 2018-03-02 at 3 22 03 pm

## How Has This Been Tested?

- [x] Unit test
- [ ] NightWatch test
- [ ] Coverage testing
- [x] Manual testing

## Test Cases and test paths

Manual test only. Test included slides in different situation.
Normal Slide & Slide with wrong date information:
screen shot 2018-03-01 at 9 43 16 pm
Slides with part of information deficiency
screen shot 2018-03-01 at 9 43 23 pm
Slide with all information deficiency
screen shot 2018-03-01 at 9 43 31 pm

Result: Only slide with both title and date information can be displayed
screen shot 2018-03-01 at 9 44 22 pm

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/specs/Calendar.spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      }
+    }
+    // To check whether the one day is added correctly
+    // Should be "2018-01-02"
+    expect((vm.addOneDay(fakeSlide)).toDateString)
+      .to.equal(('2018-01-02').toDateString)
+
+    // Now testing whether the month can be changed correctly.
+    // Change the date to '2018-01-31'
+    fakeSlide.date.content = '2018-01-31'
```
Comment (peggyopal) [inline test/unit/specs/Calendar.spec.js]: Should this be using `$data`?

Comment (Yuecheng-Rong) [reply test/unit/specs/Calendar.spec.js]: No……there's no "Data" function in the original calendar view at all. The data here is just created by this test component so $data is not necessary here

Comment (peggyopal) [reply test/unit/specs/Calendar.spec.js]: Thanks!

Comment (Yuecheng-Rong) [reply test/unit/specs/Calendar.spec.js]: You are welcome!

**Pull request merged**